### DISCUSSION TITLE
FeatureRequest: memoryview Response

### DISCUSSION THREAD

**Discussion by j-carson** [encode/starlette#1273] [Ideas] [answered]
I am assembling a large `Response` content buffer using `BytesIO`. In order to pass my data to the `Response` class, I need to make a `bytes` copy of the buffer. It would be more efficient if the `Response` class would accept a `memoryview`, so could use the `getbuffer` method and avoid this extra copy.

**Comment by bschoenmaeckers** [comment, marked as answer]
This will be possible after https://github.com/encode/starlette/pull/2577 is merged
